internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybridStoreDeletion.java

```diff
@@ -0,0 +1,213 @@
+    serverProperties.setProperty(MIN_CONSUMER_IN_CONSUMER_POOL_PER_KAFKA_CLUSTER, "1");
+    serverProperties.setProperty(SERVER_DEDICATED_DRAINER_FOR_SORTED_INPUT_ENABLED, "true");
+
+    // Set the max number of partitions to 1 so that we can test the partition-wise shared consumer assignment strategy.
+    serverProperties.setProperty(DEFAULT_MAX_NUMBER_OF_PARTITIONS, "1");
+    serverProperties.setProperty(
+        SERVER_SHARED_CONSUMER_ASSIGNMENT_STRATEGY,
+        KafkaConsumerService.ConsumerAssignmentStrategy.PARTITION_WISE_SHARED_CONSUMER_ASSIGNMENT_STRATEGY.name());
+
+    for (int i = 0; i < NUMBER_OF_SERVERS; i++) {
+      cluster.addVeniceServer(new Properties(), serverProperties);
+    }
+
+    return cluster;
+  }
+
+  /**
+   * testHybridStoreRTDeletionWhileIngesting does the following:
+   *
+   * 1. Set up a Venice cluster with 1 controller, 1 router, and 1 server.
+   * 2. Limit the shared consumer thread pool size to 1 on the server.
+   * 3. Create two hybrid stores.
+   * 4. Produce to the rt topic of the first store and allow the thread to produce some amount of data.
+   * 5. Delete the rt topic of the first store.
+   * 6. Produce to the rt topic of the second store with 10 key-value pairs.
+   * 7. Check that the second store has all the records.
+   */
+  @Test(timeOut = 120 * Time.MS_PER_SECOND)
+  public void testHybridStoreRTDeletionWhileIngesting() {
+    parentZk = ServiceFactory.getZkServer();
+    parentController = ServiceFactory.getVeniceController(
+        new VeniceControllerCreateOptions.Builder(
+            veniceCluster.getClusterName(),
+            parentZk,
+            veniceCluster.getPubSubBrokerWrapper())
+                .childControllers(new VeniceControllerWrapper[] { veniceCluster.getLeaderVeniceController() })
+                .build());
+
+    long streamingRewindSeconds = 25;
+    long streamingMessageLag = 2;
+    final String storeNameFirst = Utils.getUniqueString("hybrid-store-test-first");
+    final String storeNameSecond = Utils.getUniqueString("hybrid-store-test-second");
+    final String[] storeNames = new String[] { storeNameFirst, storeNameSecond };
+
+    try (
+        ControllerClient controllerClient =
+            new ControllerClient(veniceCluster.getClusterName(), parentController.getControllerUrl());
+        AvroGenericStoreClient<Object, Object> clientToSecondStore = ClientFactory.getAndStartGenericAvroClient(
+            ClientConfig.defaultGenericClientConfig(storeNameSecond)
+                .setVeniceURL(veniceCluster.getRandomRouterURL()))) {
+
+      createStoresAndVersions(controllerClient, storeNames, streamingRewindSeconds, streamingMessageLag);
+
+      // Produce to the rt topic of the first store and allow the thread to produce some amount of data.
+      produceToStoreRTTopic(storeNameFirst, 200);
+
+      // Delete the rt topic of the first store.
+      controllerClient.deleteKafkaTopic(Version.composeRealTimeTopic(storeNameFirst));
```

Comment:
Did it work? Can we use `assertCommand` to make sure?

Also, I believe this ends up calling `admin.truncateKafkaTopic(topicName)`, which asynchronously kicks off the deletion process by the `TopicCleanupService`, rather than deleting immediately. Moreover, there may be safeguards in the `TCS` preventing the deletion of a RT topic from a non-deleted hybrid store (not sure).

We should add a non-deterministic assertion which waits until the topic is actually deleted. Or maybe just use the Kafka tooling to directly delete the topic, rather than go through the controller client, controller, `TCS`...

Reply:
> Did it work? Can we use assertCommand to make sure?

I think so, for example, I can find the progress of deleting the rt topic from test logs. Of course we can use `assrtCommand` as suggested.
```
2024-01-23 10:50:06 - [] INFO [TopicCleanupServiceForParentController] [TopicCleanupTask] Retention policy for topic: hybrid-store-test-first_264c8084685a2_a9986bb4_rt is: 5000 ms, and it is deprecated, will delete it after 200 ms.
...
2024-01-23 10:50:06 - [] INFO [TopicManager [localhost:57659]] [TopicCleanupTask] Deleting topic: hybrid-store-test-first_264c8084685a2_a9986bb4_rt
...
2024-01-23 10:50:06 - [] INFO [TopicManager [localhost:57659]] [TopicCleanupTask] Topic: hybrid-store-test-first_264c8084685a2_a9986bb4_rt has been deleted
```
After the topic had been deleted, I was able to see e.g. similar error messages in the shared consumer thread.  
```
2024-01-23 10:50:08 - [testRegion-VeniceServer-57714] WARN [NetworkClient] [venice-shared-consumer-for-localhost:57659-t1] [Consumer clientId=localhost_localhost:57659_0, groupId=null] Error while fetching metadata with from source 1001 with correlation id 114 : {hybrid-store-test-first_264c8084685a2_a9986bb4_rt=UNKNOWN_TOPIC_OR_PARTITION}
2024-01-23 10:50:08 - [testRegion-VeniceServer-57714] WARN [NetworkClient] [venice-shared-consumer-for-localhost:57659-t1] [Consumer clientId=localhost_localhost:57659_0, groupId=null] Error while fetching metadata with from source 1001 with correlation id 117 : {hybrid-store-test-first_264c8084685a2_a9986bb4_rt=UNKNOWN_TOPIC_OR_PARTITION}
```

Reply:
So... it sounds like there is no issue with a topic getting deleted while a consumer is subscribed to it. That is fine I guess (though unexpected). I think we should still have a stronger assertion that the topic is actually deleted, e.g. via `PubSubAdminAdapter::containsTopic`, before moving on to the next step.

Once we have that, then we could merge the PR. WDYT?

Reply:
That's also my understanding from this test, our system seems to be able to handle it correctly. However, there might be some other factors that in combination with a deleted RT topic to cause the dead consumer thread issue (thus not able to ingest data), and we are not quite clear on what the other factor is.

I agree and I will add the assertion in the test to make sure the topic is deleted before the next step.